observability/WFLY-14947-Implement_observability_policy_metrics.adoc

```diff
@@ -0,0 +1,71 @@
+ = Implement Observability Policy for Metrics
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+While WildFly/EAP already supports basic metrics in the base WildFly/EAP system, as well as MicroProfile Metrics in the WildFly MicroProfile configuration and EAP XP, there is an effort in Red Hat Middleware Runtimes to implement a consistent observability policy across all Runtime offerings. In the Java space, the library of choice, and the emerging de facto standard for Java developers is https://micrometer.io[Micrometer].  This RFE will add support for Micrometer to WildFly/EAP and adhere to the policy decided upon by the various stakeholders.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14947
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1686
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [x] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* Feature parity with existing "base metrics"
+* Application registry available during injection
+* Prometheus endpoint that provides application and jvm statistics as allowed by RBAC (if enabled)
+* Deprecate/remove/replace existing base metrics?
+
```

Comment:
What requirements are there related to output format, particularly metric naming. AIUI what we do now for vendor metrics is not what is wanted, so we need requirements re what is wanted.

Reply:
The observability policy simply states "Every runtime should expose a Prometheus endpoint", which would imply that we adhere to this: https://prometheus.io/docs/practices/naming/ As I understand things, Micrometer's Prometheus exporter covers some of that, though there are parts (e.g., "must have a single unit (i.e. do not mix seconds with milliseconds, or seconds with bytes)") that may be harder to enforce?